packages/core/upload/server/graphql.js

```diff
@@ -48,7 +49,11 @@ module.exports = ({ strapi }) => {
     const currentFile = uploadService.formatFileInfo(
       {
         filename,
-        type: mimetype,
+        /**
+         * in case the mime-type wasn't sent, Strapi tries to guess it
+         * from the file extension, to avoid a corrupt database state
+         */
+        type: mimetype || mime.lookup(filename) || 'application/octet-stream',
```

Comment:
If we do it here should we still do it in the middleware ?

Reply:
Unfortunately, we do, because for `/graphql` we do have an early return, which prevents us from running this beforehand. I'm not sure what would happen if we move the fix there too tbh, but I guess there is a reason for it :)

https://github.com/strapi/strapi/blob/e297cb7c624acc35663602bc426dae3231c81b88/packages/core/strapi/lib/middlewares/body.js#L31-L33

Reply:
Ok I remembered you mentioned sth like that thanks!